src/store/confirmedTransactionsSlice.ts

```diff
@@ -50,6 +52,8 @@ const confirmedTransactionsSlice = createSlice({
       .addCase(syncAddressesData.fulfilled, addTransactions)
       .addCase(syncAddressTransactionsNextPage.fulfilled, addTransactions)
       .addCase(syncAllAddressesTransactionsNextPage.fulfilled, addTransactions)
+      .addCase(walletLocked, () => initialState)
+      .addCase(walletSwitched, () => initialState)
```

Comment:
This is the reason why I wanted to differentiate between `walletUnlocked` and `walletSwitched`. If I didn't, then we'd need to have here:
```ts
      .addCase(walletLocked, () => initialState)
      .addCase(walletUnlocked, () => initialState)
```

which is quite weird 🤔 why the hell do I need to clean the state when locking AND when unlocking?! The reason is that unlocking can happen when we are already unlocked and we switch a wallet. Thus, I made a new action for clarity.